delivery_postlogistics/models/stock_picking.py

```diff
@@ -236,3 +236,9 @@ def generate_postlogistics_shipping_labels(self, package_ids=None):
         """Add label generation for PostLogistics"""
         self.ensure_one()
         return self._generate_postlogistics_label(package_ids=package_ids)
+
+    def action_generate_carrier_label(self):
+        self.ensure_one()
+        if not self.carrier_id:
+            raise exceptions.UserError(_("Please, set a carrier."))
+        self.env["delivery.carrier"].postlogistics_send_shipping(self)
```

Comment:
what if carrier isn't postlogistics?

Reply:
I'm more concerned about carrier calling `_set_a_default_package`
https://github.com/OCA/delivery-carrier/blob/16.0/delivery_postlogistics/models/delivery_carrier.py#L139C18-L139C40

Reply:
hello @mmequignon this comes from `base_delivery_carrier_label` that has made packages mandatory since its [12.0 migration](https://github.com/Timo17100-c2c/delivery-carrier/pull/1); `delivery_postlogistics` was created based on `delivery_carrier_label_postlogistics` 12.0 (-> `base_delivery_carrier_label`) and kept same logic